source/Audio.cpp

```diff
@@ -260,6 +267,55 @@ void Audio::Update(const Point &listenerPosition)
 
 
 
+void Audio::UpdateMusic(PlayerInfo &player, Track::GameState state)
+{
+	if(!isInitialized)
+		return;
+	
+	if(Preferences::Has("ingame music"))
+	{
+		bool currentPlaylistValid = currentPlaylist ? currentPlaylist->MatchingConditions(player) : false;
+		if(currentTrack->IsFinished() || !currentPlaylistValid)
+		{
+			if(!currentPlaylistValid)
+			{
+				WeightedList<Playlist *> validPlaylists;
+				int priority = 0;
+				for(auto &playlist : GameData::Playlists())
+					if(playlist.second.MatchingConditions(player))
+					{
+						if(playlist.second.Priority() == priority)
+							validPlaylists.emplace_back(playlist.second.Weight(), &playlist.second);
+						else if (playlist.second.Priority() > priority)
+						{
+							priority =  playlist.second.Priority();
+							validPlaylists.clear();
+							validPlaylists.emplace_back(playlist.second.Weight(), &playlist.second);
+						}
+					}
+				if(validPlaylists.size())
```

Comment:
best to check if it isnt empty than to query the size just to know if its empty (I think it has a IsEmpty() method?)

Reply:
Ok

Reply:
done